src/app/components/hmcts-global-header/hmcts-global-header.component.ts

```diff
@@ -25,14 +26,52 @@ export class HmctsGlobalHeaderComponent {
 
   public showItems: boolean;
   public userValue = true;
+  public tab;
 
-  constructor(public nocStore: Store<fromNocStore.State>) { }
+  constructor(
+    public nocStore: Store<fromNocStore.State>,
+    private readonly router: Router) { }
+
+  public ngOnInit() {
+    // set the active tab via the url in the router
+    this.tab = this.setActiveTab(this.router.url);
+  }
+
+  public setActiveTab(url: string) {
+    this.tab = null;
+    switch (url) {
+      case '/tasks/list': {
```

Comment:
This was something I did. The active tab wasn't being set correctly when navigating so not sure whether the mechanism you're talking about wasn't working correctly or we weren't actually using the mechanism for some reason. If this is going to break then could just get active tab from router on changes?

Reply:
Yes, I put something in that already set the `active` state on each item. If there needed to be a change made because that wasn't working quite right, it should be done in `src/app/app-utils.ts: setActiveLink()`, but this was working so what's changed to stop that from being the case?

If it's that url is no longer a perfect match (which would make sense for certain actions), it can be done with `currentUrl.indexOf(item.href) === 0` so it still works for deeper links. But the code in this method is only doing perfect matching anyway.

This was the PR, which has already been merged into `master`: https://github.com/hmcts/rpx-xui-webapp/pull/738/files

Maybe you didn't have it on the branch?

Reply:
I don't think I saw that on the branch while I was working on it. However, there are additional actions being added to the url so that might also be why the tabs started acting up. Maybe if I were to branch out from this (as it's almost ready to be merged into master) and remove/revert these changes to see if the tabs still have issues? If they do and the setActiveLink() code is present, then I could work on the fix you've mentioned?

Reply:
Resolving this on the understanding that the change will be made in a subsequent PR.